pvlib/mlfm.py

```diff
@@ -0,0 +1,804 @@
+                 dstack['i_mp'],
+                 dstack['i_sc'] * (not is_i_sc_self_ref)]
+        labels = [
+            'pr_dc',
+            'stack_t_mod',
+            'stack_v_oc',
+            'stack_v_mp',
+            '- - -',
+            'stack_i_mp',
+            'stack_i_sc']
+        color_map = [
+            'white',  # colour to bottom of graph
+            clr['temp_module'],
+            clr['v_oc'],
+            clr['v_mp'],
+            clr['i_v'],
+            clr['i_mp'],
+            clr['i_sc']]
+
+    # offset legend right, use ~1.2
+    bbox = 1.2
+
+    # select x axis usually date_time
+    xdata = dmeas.index.values
+    fig, ax1 = plt.subplots()
+
+    ax1.set_title(title)
+
+    # plot stack in order bottom to top,
+    # allowing self_ref and temp_module corrections
+    ax1.stackplot(xdata, *tuple(ydata), labels=labels, colors=color_map)
+
+    ax1.axhline(y=1/fill_factor, c='grey', lw=3)  # show initial 1/FF
+    ax1.axhline(y=1, c='grey', lw=3)  # show 100% line
+    ax1.set_ylabel('stacked mlfm losses')
+
+    # find number of x date values
+    x_ticks = dmeas.shape[0]
+    plt.xticks(np.arange(0, x_ticks), rotation=90)
+
+    # if (xaxis_labels > 0 and xaxis_labels < x_ticks):
+    if 0 < xaxis_labels < x_ticks:
+        xaxis_skip = np.floor(x_ticks / xaxis_labels)
+    else:
+        xaxis_skip = 2
+
+    #
+    xax2 = [''] * x_ticks
+    x_count = 0
+    while x_count < x_ticks:
+        if x_count % xaxis_skip == 0:
+            #
+            #  try to reformat any date indexes (not for matrices)
+            #
+            #  0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9
+            #   y y y y - m m - d d t h h : m m : s s --> yy-mm-dd hh'h'
+            #
+            try:
+                xax2[x_count] = xdata[x_count][2:13]+'h'
+            except IndexError:
+                xax2[x_count] = xdata[x_count]
+            except TypeError:  # xdata can't be subscripted
+                xax2[x_count] = xdata[0]
+
+        x_count += 1
+
+    ax1.set_xticklabels(xax2)
+    ax1.set_ylim(0.6, 1/fill_factor + 0.1)  # optional normalised y scale
+    plt.legend(bbox_to_anchor=(bbox, 1), loc='upper left', borderaxespad=0.)
+
+    # plot met data on right y axis
+    ax2 = ax1.twinx()
+    ax2.set_ylabel('poa_global (kW/m^2), temp_module (C/100)')
+    ax2.set_ylim(0, 4)  # set so doesn't overlap mlfm params
+
+    plt.plot(xdata, dmeas['poa_global'] / G_STC,
+             c=clr['irradiance'], label='poa_global (kW/m^2)')
+    plt.plot(xdata, dmeas['temp_module'] / 100,
+             c=clr['temp_module'], label='temp_module / 100')
+
+    # temp_air may not exist particularly for indoor measurements
+    try:
+        plt.plot(xdata, dmeas['temp_air']/100,
+                 c=clr['temp_air'], label='temp_air/100')
+    except KeyError:
+        pass
+
+    ax2.legend(bbox_to_anchor=(bbox, 0.3), loc='upper left', borderaxespad=0.)
+    ax1.set_xticklabels(xax2, rotation=90)
+
+    return fig
+
+
+REFS = """
```

Comment:
Where does REFS get used?

Reply:
It doesn't, that's what  @cwhanse changed it to, Cliff?

Reply:
I don't recall why or when that was changed to a constant string.